drafts/25-xxx-specifications.txt

```diff
@@ -0,0 +1,491 @@
+From: JoR
+Subject: Formal specifications for the Fortran preprocessor (FPP)
+Date: 2025-March-09
+References: 25-114r2
+            25-007 Fortran 2023 Interpretation Document
+            24-108 Preprocessor directives seen in existing Fortran
+                   programs.txt
+            24-109 On Fortran awareness in a Fortran preprocessor.txt
+            ISO/IEC 9899:2023 Programming languages -- C
+                   (working draft N3096)
+
+
+1. Introduction
+===============
+
+At its meeting Feb 19, 2025, J3 decided to approve requirements for
+a cpp-like preprocessor for Fortran 202Y (paper 25-114r2).
+
+This is the formal specifications document, revised according to the
+discussion arising from 25-114r2.
+
+Terminology: For the purpose of this specification, the 'preprocessor'
+encompasses any stages of preprocessing of the input text. For didactic
+purposes, that might include additional phases of "preprocessing" that
+help define the expected priorities of preprocessing behaviors. (In past
+discussions, these have included line continuation processing, comment
+handling, and tokenization for the preprocessor.)
+
+
+2. Formal specifications
+========================
+
+2.1 Lexical specifications
+--------------------------
+
+2.1.1 Lines
+-----------
+
+li00. The Fortran preprocessor recognizes three distinct types of lines:
+      Fortran source fragments, Fortran comment lines, and preprocessing
+      directives.
+
+li10. Fortran comment lines are defined as in 25-007 6.2.2.3 and
+      6.2.3.2.
+
+li20. Lines that are not Fortran comment lines that have a '#'
+      character as the first non-blank character of the line are
+      directive lines, as required by C 2023 6.10.1 paragraph 2.
+
+li21. In fixed-form input files, a '#' in column 6 of a non-comment
+      line does not introduce a directive line.
+
+li22. A preprocessor directive can be continued with a backslash '\'
+      immediately followed a new-line. The backslash and new-line are
+      deleted.
+
+li23. The maximum length of a preprocessor directive (including
+      continuation text) is one million characters.
+
+li24. A non-empty source file shall not end with a '\' followed
+      immediately by a new-line as required in C 2023 5.1.1.2 bullet 2.
+
+li30. Fortran source fragments are those lines that are not Fortran
+      comment lines nor preprocessor directive lines.
+
+li21. In a fixed-form Fortran fragment, a '#' in column 6 does not make
+      the line a directive line; it is a Fortran source fragment.
+
+
+
+li31. Text on fixed-form Fortran source fragments is ignored beyond
+      column 72.
+
+li32. Fortran source fragments may be continued with a continuation
+      ('&' at the end of a free-form line as specified in Fortran 2023
+      6.3.2.4, or with a non-blank, non-zero character in column 6
+      of a fixed-form line as specified in Fortran 2023 6.3.3.3).
+
+      Example 1 (free-form):
+          call subroutine foo(1, 2, &
```

Comment:
I'm unfamiliar with the use of the `subroutine` token within `call` syntax and suspect it's a syntax error.

```suggestion
          call subroutine_foo(1, 2, &
```